siriuspy/siriuspy/pwrsupply/pssofb.py

```diff
@@ -609,3 +581,200 @@ def _create_pru_udc(ethbridgeclnt_class):
+
+    def bsmp_sofb_kick_set(self, kick):
+        """Send kick sofb setpoint to power supplies."""
+        current = self.converter.conv_stren2curr(kick)
+        self.bsmp_sofb_current_set(current)
+
+    def bsmp_sofb_current_set_update(self, current):
+        """Send current sofb setpoint to power supplies and update."""
+        self._parallel_execution('bsmp_sofb_current_set_update', (current, ))
+
+    def bsmp_sofb_kick_set_update(self, kick):
+        """Send kick sofb setpoint to power supplies and update."""
+        current = self.converter.conv_stren2curr(kick)
+        self.bsmp_sofb_current_set_update(current)
+
+    def bsmp_update_sofb(self):
+        """Update SOFB currents by sending correctors bsmp requests."""
+        self._parallel_execution('bsmp_update_sofb')
+
+    def bsmp_update_state(self):
+        """Receive from correctors variables and update object attributes."""
+        self._parallel_execution('bsmp_update_state')
+
+    def bsmp_pwrstate_on(self):
+        """Turn all correctors on and then update state."""
+        self._parallel_execution('bsmp_pwrstate_on')
+
+    def bsmp_pwrstate_off(self):
+        """Turn all correctors off and then update state."""
+        self._parallel_execution('bsmp_pwrstate_off')
+
+    def bsmp_slowref(self):
+        """Set all correctors to SlowRef opmode and then update state."""
+        self._parallel_execution('bsmp_slowref')
+
+    def bsmp_slowrefsync(self):
+        """Turn all correctors ro SlowRefSync opmode and then update state."""
+        self._parallel_execution('bsmp_slowrefsync')
+
+    # --- sofb-vector object attribute methods (no comm.) ---
+
+    @property
+    def sofb_psnames(self):
+        """Return corrector names in SOFB order."""
+        return self._sofb_psnames
+
+    @property
+    def sofb_kick_readback_ref(self):
+        """Return SOFB kick from current_readback_ref from last setpoint."""
+        return self.converter.conv_curr2stren(self.sofb_current_readback_ref)
+
+    @property
+    def sofb_current_readback_ref(self):
+        """Return SOFB current readback after last setpoint."""
+        return self._sofb_current_readback_ref.copy()
+
+    @staticmethod
+    def sofb_vector_issame(vector1, vector2):
+        """Check if two sofb vectors are identical within float32 precision."""
+        if len(vector1) != len(vector2):
+            return False
+        # comparison
+        vec1 = _np.asarray(vector1, dtype=_np.float32)
+        vec2 = _np.asarray(vector2, dtype=_np.float32)
+        return _np.all(vec1 == vec2)
+
+    def conv_psname_2_sofb_index(self, psname):
+        """."""
+        return self._sofb_psnames.index(psname)
+
+    def shutdown(self):
+        """."""
+        for pipe in self._pipes:
+            pipe.send(None)
+            pipe.close()
+        for proc in self._procs:
+            proc.join()
+
+    @staticmethod
+    def run_process(ethbridgeclnt_class, bbbnames, pipe, doneevt, shared_arr):
```

Comment:
@fernandohds564 , I think this should be a private method, right?

Reply:
yes, it would be better that way

Reply:
done!